scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -738,6 +751,25 @@ function Setup-Environment([string] $ContainerOs, [switch] $SkipArchCheck) {
             [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12
     }
 
+    if (($ContainerOs -eq 'Windows') -and (-not $SkipBatteryCheck)) {
+        if ((Get-Command 'Get-WmiObject' -ErrorAction Ignore) -ne $null) {
+            [psobject[]] $batteries = Get-WmiObject Win32_Battery -ErrorAction Ignore
+            if (($batteries -ne $null) -and ($batteries.Length -gt 0)) {
+                Write-Warning (
```

Comment:
Is there a way to test for the fix so that this goes away as the fix is deployed? Because even after the fix is deployed, we will have some devices with the fix and some without.

Reply:
Once Windows releases a fix, we can enhance this to test for the build number (just like we compare against 17763 right now for Windows container support). There is no way to detect the existence of the issue up-front AFAIK.

Reply:
Just to note, the build number check will not work, as that number will not change (once RS5, stays RS5, patches are just packages installed, they do not change the build number). I think there is a way to test for a minimum patch set, so that would be ok. Sounds good, if there is no good way to detect otherwise.

Reply:
Hmm, okay. We'll revisit when the fix is released.